File: Chapter5/5.1/parity/cpp/parity.cpp
#include <iostream>
#include <bitset>

using namespace std;

short parity(long x) {
  short result = 0;
  while (x != 0) {
    result ^= 1;
    x &= x - 1;
  }
  return result;
}

int main() {
  long x = 0xF0F0000F;
  cout<<bitset<64>(x)<<" has parity "<<parity(x)<<endl;
}

File: Chapter5/5.1/parity/java/parity.java
public class parity {
    static public short parity(long x) {
	x ^= x>>32;
	x ^= x>>16;
	x ^= x>>8;
	x ^= x>>4;
	x ^= x>>2;
	x ^= x>>1;
	return (short)(x&1);
    }

    public static void main(String[] args) {
	long x = 0xFF000001;
	System.out.println("Type long has " + Long.SIZE + " bits.");
	System.out.println(Long.toBinaryString(x) + " has parity " + parity(x));
    }
}
